internal/errors/errors_test.go

```diff
@@ -0,0 +1,1498 @@
+			return test{
+				name: "return nil when a format is empty and args has multiple values.",
+				args: args{
+					args: val,
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			val := []interface{}{
+				map[string]int{"invalid time_duration": 10},
+			}
+			wantErr := fmt.Errorf("%v", val[0])
+			return test{
+				name: "return an error when a format is empty and args has a single value",
+				args: args{
+					args: val,
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := Wrapf(test.args.err, test.args.format, test.args.args...)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestCause(t *testing.T) {
+	type args struct {
+		err error
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return fmt.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		func() test {
+			wantErr := errors.Unwrap(Wrap(errors.New("err"), "invalid parameter"))
+			return test{
+				name: "return an unwrapped error when err is not empty.",
+				args: args{
+					err: Wrap(errors.New("err"), "invalid parameter"),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			return test{
+				name: "return nil when err is empty.",
+				args: args{},
+				want: want{},
+			}
+		}(),
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := Cause(test.args.err)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestUnwarp(t *testing.T) {
+	type args struct {
+		err error
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return fmt.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		func() test {
+			wantErr := errors.New("err")
+			err := fmt.Errorf("%s: %w", "error occurs", wantErr)
```

Comment:
can you tell me why you use `pkg/errors` and `fmt` at the same time here?

Reply:
@kevindiu here https://github.com/vdaas/vald/pull/929#discussion_r563425046